plugins/org.jboss.tools.jst.angularjs/src/org/jboss/tools/jst/angularjs/internal/ExpressionsProposalProcessor.java

```diff
@@ -0,0 +1,163 @@
+	 */
+	@Override
+	public TextProposal[] getProposals(KbQuery query, IPageContext context) {
+		List<TextProposal> proposals = new ArrayList<TextProposal>();
+		String qValue = query.getValue();
+		int startEl = qValue.lastIndexOf("{{");
+		if(startEl<0) {
+			qValue = query.getRegionValuePrefix();
+			if(qValue!=null) {
+				startEl = qValue.lastIndexOf("{{");
+			}
+		}
+		if(startEl>-1) {
+			String value = qValue.substring(startEl + 2);
+			if(!value.contains("\n") && !value.contains("}") && (new AngularJSRecognizer()).isUsed(context)) {
+				// Create a JS
+				String js = generateJs(value, context);
+				Collection<Object> results = BrowserDataProviderManager.getInstance().evaluate(js, context);
+				// Convert the data to a list of proposals
+				if(!results.isEmpty()) {
+					if(image == null) {
+						image = ImageDescriptor.createFromFile(AngularJsPlugin.class, "angular.png");
+					}
+					for (Object object : results) {
+						String result = object.toString();
+						StringTokenizer st = new StringTokenizer(result, ";", false);
+						int dotIndex = value.lastIndexOf('.');
+						String remove = value;
+						if(dotIndex>-1) {
+							remove = value.substring(dotIndex + 1);
+						}
+						String suffix = query.getRegionValueSufix();
+						if(suffix==null) {
+							suffix = "";
+						}
+						while(st.hasMoreElements()) {
+							String label = st.nextToken().trim();
+							if(!label.matches("\\d+")) { // Ignore numbers
+								StringBuilder replacementString = new StringBuilder();
+								if(query.getType() == KbQuery.Type.ATTRIBUTE_VALUE) {
+									replacementString.append(qValue);
+								}
+								replacementString.append(label.substring(remove.length()));
+								int position = replacementString.length();
+								if(query.getType() == KbQuery.Type.ATTRIBUTE_VALUE) {
+									replacementString.append(suffix);
+								}
+								int close = suffix.indexOf("}}");
+								boolean closedExp = false;
+								if(close>-1) {
+									String c = suffix.substring(0, close);
+									if(!c.contains("\n")) {
+										closedExp = true;
+									}
+								}
+								if(!closedExp) {
+									replacementString.append("}}");
+								}
+								TextProposal proposal = new TextProposal();
+								proposal.setLabel(label);
+								proposal.setReplacementString(replacementString.toString());
+								proposal.setPosition(position);
+								proposal.setImageDescriptor(image);
+								proposal.setAutoActivationContentAssistantAfterApplication(false);
+								proposals.add(proposal);
+							}
+						}
+					}
+				}
+			}
+		}
+		return proposals.toArray(new TextProposal[proposals.size()]);
+	}
+
+	private String generateJs(String value, IPageContext context) {
+		String js =
+				  "var ORG_JBOSS_TOOLS_JST = {};"
+
+				+ "ORG_JBOSS_TOOLS_JST.getProposals = function(element, value) {"
```

Comment:
what happens with this javascript if angular is not present ?

I would assume somewhere in this should be a guard against 'angular' not being defined to avoid unnecessary errors/warnings ?

and possibly it should be guarded even more heavily if this snippet is going to be injected into *everything* ?